### PR TITLE
INTYGFV-14459: Added resource links for user

### PR DESCRIPTION
**Pull request by mhornfeldt**

### CODE REVIEW

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Two small things to look into, but other than that it looks good!

File: web/src/main/java/se/inera/intyg/webcert/web/web/controller/facade/dto/ResourceLinkTypeDTO.java [hunk 1]
```diff
@@ -42,4 +42,6 @@
     CANNOT_COMPLEMENT_CERTIFICATE,
     CREATE_CERTIFICATE_FROM_TEMPLATE,
     CREATE_CERTIFICATE_FROM_CANDIDATE,
+    CREATE_CERTIFICATE,
```
[the remark this answers is not in the document]
Comment (mhornfeldt) [reply web/src/main/java/se/inera/intyg/webcert/web/web/controller/facade/dto/ResourceLinkTypeDTO.java:45]: Searching for a patient and creating a certificate is in the same tab in the application. If you haven't searched for a patient you won't see the create certificate section, so I was thinking that the resource link would handle both functions. I can change the name.

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/GetUserResourceLinksImpl.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import se.inera.intyg.webcert.web.service.facade.GetUserResourceLinks;
+import se.inera.intyg.webcert.web.service.user.WebCertUserService;
+import se.inera.intyg.webcert.web.service.user.dto.WebCertUser;
+import se.inera.intyg.webcert.web.web.controller.facade.dto.ResourceLinkDTO;
+import se.inera.intyg.webcert.web.web.controller.facade.dto.ResourceLinkTypeDTO;
+
+import java.util.ArrayList;
+
+@Service
+public class GetUserResourceLinksImpl implements GetUserResourceLinks {
```
[the remark this answers is not in the document]
Comment (mhornfeldt) [reply web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/GetUserResourceLinksImpl.java:32]: When you say have the user as an argument do you mean of class User or WebCertUser? If we don't use the WebCertUser coming from the webcertUserService we don't get access to the origin of the user.

File: web/src/main/java/se/inera/intyg/webcert/web/web/controller/facade/dto/ResourceLinkTypeDTO.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mww-sopra-steria) [reply web/src/main/java/se/inera/intyg/webcert/web/web/controller/facade/dto/ResourceLinkTypeDTO.java:45]: Sounds good that it handles both. It makes sense. But maybe think about a name-pattern that also makes sense for the other "tabs" like 'Signerade intyg' etc.

**Review by mww-sopra-steria — APPROVED** [review]

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
It looks good! I like that you have created a ResourceLinkFacadeTestHelper! Could you see if there are more test classes that could be refactored to use the helper? For example GetQuestionsAvailableFunctions.

The changes in UserServiceImpl looks good, but could you limit to SITHS, FAKE and BANK_ID? We don't support any other login methods in the React-client and it is probably better not to expose anything else to the client.

**Review by mikeyOne — APPROVED** [review]

**Pull request merged**